src/NATS.Client.JetStream/Internal/NatsJSConsume.cs

```diff
@@ -128,11 +128,12 @@ internal class NatsJSConsume<TMsg> : NatsSubBase
             Timeout.Infinite,
             Timeout.Infinite);
 
-        // This channel is used to pass messages
-        // to the user from the subscription channel (which should be set to a
-        // sufficiently large value to avoid blocking socket reads in the
-        // NATS connection).
-        _userMsgs = Channel.CreateBounded<NatsJSMsg<TMsg>>(1000);
+        // This channel is used to pass messages to the user from the subscription channel.
+        // Capacity is set to maxMsgs to avoid pulling too many messages from the server
+        // resulting in a large number of pending messages (which can be a problem if the
+        // application is slow to process messages). The capacity is bounded to a maximum of 1024
+        // to avoid LOH allocations.
+        _userMsgs = Channel.CreateBounded<NatsJSMsg<TMsg>>(maxMsgs > 1024 ? 1024 : (int)maxMsgs);
```

Comment:
Is there ever a case where max msgs is 0(or negative), such as if they only specify max bytes?

Reply:
Good point. should we set a minimum?

Reply:
Maybe we should just make sure it’s at least 1? If <=0 default it to 1024, because that probably means they are setting max bytes and not max msgs

Reply:
makes sense

Reply:
changing the implementation to use 'Delivered()' function. this is set to a fixed 1024 since the size of the channel will not affect the pull requests anymore.